heltour/tournament/tasks.py

```diff
@@ -311,6 +317,99 @@ def update_slack_users():
             p.save()
 
 
+def _start_league_games(*, tokens, clock, increment, do_clockstart, clockstart, clockstart_in, variant, leaguename, league_games):
+    try:
+        result = lichessapi.bulk_start_games(tokens=tokens, clock=clock, increment=increment, do_clockstart=do_clockstart, clockstart=clockstart, clockstart_in=clockstart_in, variant=variant, leaguename=leaguename)
+    except lichessapi.ApiClientError as err:
+        # try to handle errors due to rjected tokens
+        e = str(err).replace('API failure: CLIENT-ERROR: [400] ', '') # get json part from error
+        try:
+           result = json.loads(e)
+           for bad_token in result["tokens"]:
+               # set expiration of rejected token to yesterday, so we know to not use it anymore.
+               for game in league_games:
+                   if game.white.oauth_token.access_token==bad_token:
+                       game.white.oauth_token.expires = timezone.now() + timedelta(days=-1)
+                       game.white.oauth_token.save()
+                       break # only one oauth_token can be the bad token, so we do not need to proceed the loop
+                   if game.black.oauth_token.access_token==bad_token:
+                       game.black.oauth_token.expires = timezone.now() + timedelta(days=-1)
+                       game.black.oauth_token.save()
+                       break
+               # remove bad token from our token string + the good token paired with it, remove potential superfluous comma
+               new_tokens = re.sub('^,', '', re.sub(f'(^|,)([A-z0-9_]*:)?{bad_token}(:[A-z0-9_]*)?', '', tokens))
```

Comment:
I'm a fan of refactoring out the various patterns in a regex expression and naming them appropriately. I find it makes it easier to figure out what is going on.